text/1001-edgedb-install.rst

```diff
@@ -0,0 +1,298 @@
+RFC 1001: CLI for installation and control of local EdgeDB server
+=================================================================
+
+This RFC describes the design of the ``edgedb`` CLI subcommand for the
+purposes of installation, update and control of a local EdgeDB server.
+
+
+Motivation
+==========
+
+Currently, the tasks of installing, updating and running an EdgeDB server
+instances are entirely manual and vary a lot across the supported platforms.
+From the standpoint of local development this creates unnecessary friction
+and opens lots of possibilities for user error.  The current state also
+necessiates a non-trivial amount of documentation that new users must read.
+
+By implementing the installation, update, and control logic into the ``edgedb``
+CLI we can significantly simplify the process of getting started with EdgeDB
+without leaving the familiar development process:
+
+1. Download the ``edgedb`` binary via the most convenient channel
+   (`npm`, `pip`, `curl`).
+2. Use the downloaded ``edgedb`` binary to either install and initialize
+   a local server, or configure a remote server instance for development.
+
+
+Overview
+========
+
+The RFC proposes a new group of ``edgedb`` CLI commands under ``edgedb server``
+prefix:
+
+* ``edgedb server install`` -- install or update a specific version of the
+  EdgeDB server on the local machine;
+
+* ``edgedb server uninstall`` -- uninstall a specific version of the
+  EdgeDB server or all versions of EdgeDB from the local machine;
+
+* ``edgedb server start`` -- starts an EdgeDB server instance;
+
+* ``edgedb server status`` -- show the status of the local EdgeDB server;
+
+* ``edgedb server stop`` -- stop the given EdgeDB server instance;
+
+* ``edgedb server upgrade`` -- upgrade the specified EdgeDB server instance
+  to the new major version.
+
+* ``edgedb server prune`` -- removes upgrade backups and other unused data.
+
+Common aspects of the implementation of the above commands is discussed
+below in `design-considerations`_.
+
+
+edgedb server install
+=====================
+
+Downloads and installs a given EdgeDB server version
+(latest stable by default).
+
+Arguments
+---------
+
+``--version=<ver>``
+  specifies the major version of the server to install.
+
+``--nightly``
+  if passed, the latest nightly build from the specified version channel
+  is installed.
+
+``--user``
+  install into the user home directory instead of system-wide, does not
+  require privelege elevation (TODO: this requires relocatable
+  builds and is TBD).
+
+``--list``
+  list EdgeDB versions available for installation
+
+``--list-installed``
+  list installed EdgeDB versions
+
+``--update``
+  if specified, ``edgedb install`` will only attempt to update the existing
+  installations.
+
+``--docker``
+  Use Docker instead of downloading and installing packages directly onto
+  the user's system.  The Docker daemon must be present and accessible by
+  the user.
```

Comment:
Should this be the default? I'd add the explanation why it's not the default to the design considerations section.

Reply:
I would not make this a default, because that is a hefty dependency to rely on.